src/test/scala/org/broadinstitute/hail/utils/OrderedRDDSuite.scala

```diff
@@ -151,10 +158,29 @@ class OrderedRDDSuite extends SparkSuite {
         .collect()
         .forall { case (v1, v2) => v1 == v2 }
     }
+
+    val scrambled = for (rdd <- Gen.oneOfGen(sorted, locusSorted);
+      newPartitions <- Gen.shuffle(rdd.partitions.indices)
+    ) yield (rdd, rdd.reorderPartitions(newPartitions.toArray))
+
+    property("partitionReordering") = Prop.forAll(scrambled) { case (original, scrambled) =>
+
+      val (status, result) = OrderedRDD.coerce(original)
+      val (status2, result2) = OrderedRDD.coerce(scrambled)
+
+      status == status2 && result.fullOuterJoin(result2)
+        .forall { case (v, (s1, s2)) => s1 == s2 }
```

Comment:
This makes me wish the quick check library had a way to point out what lead to the falsification of the property. Here we really have a huge number of conjunctions, and it might not be obvious which one is causing the problem.